modules/guides/pages/javascript-udfs/creating-javascript-udf.adoc

```diff
@@ -0,0 +1,56 @@
+= Creating a Javascript UDF
+:description: Using the Couchbase Query Workbench to create a user-defined function written in Javascript
+:page-pagination: 
+:page-edition: Enterprise Edition
+:page-topic-type: guide
+:page-toclevels: 2
+
+[abstract]
+{description}
+
+== Introduction
+
+We're going to introduce a number of new concepts in this example:
+
+* Passing parameters to Javascript UDFs.
+* Returning values from Javascript UDFs.
+
+[#introduction-create-javascript-library]
+== Creating a Javascript Library
+
+First, you need to create a library to store your Javascript functions.
+
+Select btn:[Query] to access the Query Workbench, then select btn:[UDF] Query Workbench menu.
+
+image::javascript-udfs/navigate-to-udf-query.png[alt="route to the user-defined functions screen"]
+
+You are now in the User-Defined Functions screen. 
+
+image::javascript-udfs/udf-library-screen.png[alt="user-defined functions library screen"]
+
+To create a new function library, click on the *{plus} add function library* link in the `Javascript Function Libraries` table.
+
+The `add library` screens is displayed.
+
+image::javascript-udfs/add-library-screen.png[,600, alt="add screen"]
+
+You have to give the library a name before you can save it; in this example, we've simply called it `my-library`.
+New libraries are created with a simple `add` example function to start with.
+
+You can add your own function, such as the one given below, which will find the number of business days (weekdays) between two dates.
+
+
+[source, javascript]
+----
+include::example$javascript-udfs/get-business-days.js[]
+----
+
+image::javascript-udfs/added-get-business-days-function.png[,600, alt="added new function"]
+
+Now, save the library by pressing the btn:[Save] button.
+
+Your library is created and will show up on the `Javascript Function Libraries`list.
+
+image::javascript-udfs/my-library-list.png[]
```

Comment:
First thing is, this only covers the UI. In all of the other how-to guides we've covered the UI and the programmatic methods, given that the how-to guides are aimed at developers.

Secondly, need to tighten up the wording to make it into more of a procedure than a tutorial. I've had a stab at rewording it below.

```suggestion
[{tabs}]
====
Query Workbench::
+
--
To create a library to store your JavaScript functions:

1. Select menu:Query[UDF] to display the User-Defined Functions screen.
+
image::javascript-udfs/udf-library-screen.png[alt="User-defined functions library screen"]

// no need to show images of how to select the User-Defined functions screen.
// This should be covered in the Query Workbench documentation by DOC-9596.

2. In the `Javascript Function Libraries` table, click on the *{plus} add function library* link.
+
The `add library` screens is displayed.
New libraries are created with a simple `add` example function to start with.

3. In the *Library Name* box, enter a name for the library.
+
image::javascript-udfs/add-library-screen.png[,600, alt="add screen"]

4. Enter your own JavaScript functions as required.
+ 
image::javascript-udfs/added-get-business-days-function.png[,600, alt="added new function"]

5. When you are ready, save the library by pressing the btn:[Save] button.
+
Your library is created and will show up on the `Javascript Function Libraries`list.
+
image::javascript-udfs/my-library-list.png[]

'''

For example, the function given below finds the number of business days (weekdays) between two dates.

[source, javascript]
----
include::example$javascript-udfs/get-business-days.js[]
----
--
====
```

Reply:
Note the main difference — in a tutorial, we tell the users what to type. In a how-to guide, we tell them where the commands are, and give them examples if appropriate; but they type what they want.

Reply:
All this has been revamped.  Needs reviewing again.